lifecycle-operator/controllers/lifecycle/keptnapp/controller_test.go

```diff
@@ -91,6 +93,69 @@ func TestKeptnAppReconciler_createAppVersionSuccess(t *testing.T) {
 
 }
 
+func TestKeptnAppReconciler_createAppVersionError(t *testing.T) {
+
+	app := &lfcv1beta1.KeptnApp{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       "my-app",
+			Namespace:  "default",
+			Generation: 1,
+		},
+		Spec: lfcv1beta1.KeptnAppSpec{
+			Version: "1.0.0",
+		},
+		Status: lfcv1beta1.KeptnAppStatus{},
+	}
+	appContext := &lfcv1beta1.KeptnAppContext{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       "my-app-context",
+			Namespace:  "default",
+			Generation: 1,
+		},
+		Spec: lfcv1beta1.KeptnAppContextSpec{
+			DeploymentTaskSpec: lfcv1beta1.DeploymentTaskSpec{
+				PreDeploymentTasks: []string{
+					"some-pre-deployment-task1",
+				},
+				PostDeploymentTasks: []string{
+					"some-post-deployment-task2",
+				},
+				PreDeploymentEvaluations: []string{
+					"some-pre-evaluation-task1",
+				},
+				PostDeploymentEvaluations: []string{
+					"some-pre-evaluation-task2",
+				},
+			},
+			Metadata: map[string]string{
+				"test1": "test2",
+			},
+			SpanLinks: []string{
+				"spanlink1",
+			},
+		},
+		Status: lfcv1beta1.KeptnAppContextStatus{},
+	}
+	r, _ := setupReconciler(app, appContext)
+
+	fakeClient := k8sfake.NewClientBuilder().WithScheme(scheme.Scheme).WithInterceptorFuncs(interceptor.Funcs{
+		Create: func(ctx context.Context, client client.WithWatch, obj client.Object, opts ...client.CreateOption) error {
+			return errors.New("unexpected error")
+		},
+	}).WithObjects(app, appContext).Build()
+
+	r.Client = fakeClient
+
+	_, err := r.Reconcile(context.TODO(), ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: app.Namespace,
+			Name:      app.Name,
+		},
+	})
+	require.NotNil(t, err)
```

Comment:
let's add more assertions here, we should expect a exact error here. Asserting for some error is little bit too general and can lead to a non-expected error and bugs